### ISSUE TITLE
MMS not imported when SMSSecure import SMS database

### ISSUE THREAD

**Issue by Informartiste** [SMSSecure/SMSSecure#140]
Hello,
I was using Hangout to handle my SMS, I have many years of history.
I installed SMSSecure 0.8.1 from Google Play and imported the SMS database.

Then I noticed I lost every MMS I previously had. The MMS are still in the Hangout database but not in the SMSSecure database.

I am able to send and receive MMS with SMSSecure, but no MSM were imported when I imported the SMS database.

THank you for your help.

Frédéric.

**Comment by twprogrammers** [comment]
+1

**Comment by breversa** [comment]
Same problem with SMSsecure 0.10.1.

**Comment by ArchangeGabriel** [comment]
This is one of the major issue of this app for me, together with MMS not being exported in backup (probably for the same reasons).

**Comment by ArchangeGabriel** [comment]
@BastienLQ Will this be solved with incoming backup/restore functionality, or is it a separated task?

**Comment by BastienLQ** [comment]
It's a different issue, because it is when Silence reads system database, not when it reads its own internal database.

**Comment by ArchangeGabriel** [comment]
OK, and what about “SMS Backup & Restore” MMS? (i.e. the currently used format for export and import) I still have my old phone backup around, because they weren’t imported. But now that I think about it, I restored it into the system database and then into (formerly) SMSSecure.

Is MMS restoring from “SMS Backup & Restore“ format working fine? If so, that’s also a workaround for this topic issue, because you could then make a backup from system database using the corresponding app, and thin import it in Silence.

**Comment by BastienLQ** [comment]
@ArchangeGabriel I don't know as I have never tested this.

**Comment by ArchangeGabriel** [comment]
OK, I’m going to test this and report.

**Comment by ArchangeGabriel** [comment]
Doesn’t work. Also, note that Silence “Import unencrypted backup” (not sure about the exact wording, “Importer la sauvegarde non-chiffrée” in french) doesn’t allow specifying the file (I think it should), I had to rename it to SilencePlaintextBackup.xml.

So, the final word is that Silence can’t import or export MMS from/to either system database (well it doesn’t to any export to system database at all for what matters, since the option does not exist) or “SMS Backup & Restore” file format.

**Comment by axelsimon** [comment]
Doesn't the encrypted export that appeared a few months back solve this issue? From what I understand, it basically exports the entire /data/data/org.smssecure.smssecure to the SD card, from where it can then be moved, copied, reimported, etc.

**Comment by ArchangeGabriel** [comment]
@axelsimon: Not at all. The issue here is importing pre-existing database, either the system one or one using the “SMS Backup & Restore” format.

**Comment by miko69** [comment]
Still no solution to import / export MMS?

**Comment by ArchangeGabriel** [comment]
Nope…
